sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -903,6 +903,8 @@ public void processElement(ProcessContext context) throws Exception {
       if (connection == null) {
         connection = dataSource.getConnection();
       }
+      // PostgreSQL requires autocommit to be disabled to enable cursor streaming
```

Comment:
drive-by: could you include the Postgres documentation link in this comment?

Reply:
Do you mean the comment in the source code or the comment of the commit?

Reply:
In the source code. If someone wonders why this is here and why Postgres requires this (and whether other DBs might also require it for the same reason), it's better if they can see it right there in the code rather than have to dig into Git history.

I would perhaps propose the following phrasing:

```
Some databases require autocommit to be disabled to enable cursor streaming.
E.g. PostgreSQL: https://jdbc.postgresql.org/documentation/head/query.html#query-with-cursor
```

Reply:
Done meanwhile ;)

Reply:
I'd suggest to set it to `false` only if it was not explicitly set to `true` by connection properties configuration. Otherwise, it won't be possible to override and it can be confusing for users.

Reply:
So we would store the `config`parameter passed in `withDataSourceConfiguration`? What would then the path to retrieve this parameter from it?

Reply:
Well, I guess it won't be so easy to check without additional work (please, correct me if I'm wrong). 

My main concern is the following - if we provide a way to configure IO in some way, we don't have to silently override it without taking into account the user config options if they were set explicitly.

Reply:
If I have an idea on how the parameter can be specified, it will be additional work yes, but I can spend some reasonable time on it.

However, I am inclined to invoke YAGNI principle: once this parameter is needed (and I can't determine how someone would need autocommit while reading), the way to parameterize it and to take it into account in `JdbcIO` will appear more clearly.

Reply:
Ok, since we have an agreement in general, let's keep this behaviour but, please, add a log message that `autoCommit` was forced to `true` and, as I mentioned before, create PR from a feature branch. Thanks!